tests/chainer_tests/functions_tests/noise_tests/test_simplified_dropconnect.py

```diff
@@ -0,0 +1,105 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer import cuda
+from chainer import functions
+from chainer.functions.noise import simplified_dropconnect
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+
+
+@testing.parameterize(*testing.product({
+    'x_dtype': [numpy.float16, numpy.float32, numpy.float64],
+    'W_dtype': [numpy.float16, numpy.float32, numpy.float64],
+    'ratio': [0.0, 0.9],
+    'train': [True, False],
+}))
+class TestSimplifiedDropconnect(unittest.TestCase):
+
+    def setUp(self):
+        self.W = numpy.random.uniform(
+            -1, 1, (2, 3)).astype(self.W_dtype)
+        self.b = numpy.random.uniform(
+            -1, 1, 2).astype(self.x_dtype)
+
+        self.x = numpy.random.uniform(-1, 1, (4, 3)).astype(self.x_dtype)
+        self.gy = numpy.random.uniform(-1, 1, (4, 2)).astype(self.x_dtype)
+        self.y = self.x.dot(self.W.T) + self.b
+        self.check_forward_options = {}
+        self.check_backward_options = {}
+        if self.x_dtype == numpy.float16:
+            self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-2}
+            self.check_backward_options = {
+                'dtype': numpy.float64, 'atol': 5e-4, 'rtol': 5e-3}
+        elif self.W_dtype == numpy.float16:
+            self.check_backward_options = {
+                'dtype': numpy.float64, 'atol': 5e-4, 'rtol': 5e-3}
+
+    def check_forward(self, x_data, W_data, b_data):
+        # Check only data type, y is tested by SimplifiedDropconnect link test.
+        x = chainer.Variable(x_data)
+        W = chainer.Variable(W_data)
+        if b_data is None:
+            y = functions.simplified_dropconnect(x, W, None, self.ratio,
+                                                 self.train)
+        else:
+            b = chainer.Variable(b_data)
+            y = functions.simplified_dropconnect(x, W, b, self.ratio,
+                                                 self.train)
+        self.assertEqual(y.data.dtype, self.x_dtype)
+
+    @condition.retry(3)
```

Comment:
We do not have to retry because the main purpose of `retry` is to reduce unexpected failures caused by numerical instability.